examples/api/lib/material/selection_area/selection_area.1.dart

```diff
@@ -0,0 +1,301 @@
+
+class SelectionAreaComplexEmphasizeTextExampleApp extends StatelessWidget {
+  const SelectionAreaComplexEmphasizeTextExampleApp({super.key});
+
+  // This widget is the root of your application.
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      theme: ThemeData(
+        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
+        useMaterial3: true,
+      ),
+      home: const MyHomePage(title: 'Flutter Demo Home Page'),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  const MyHomePage({super.key, required this.title});
+
+  final String title;
+
+  @override
+  State<MyHomePage> createState() => _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  final ContextMenuController _menuController = ContextMenuController();
+  final SelectionController _selectionController = SelectionController();
+  final int _text1Id = SelectableRegionState.nextSelectableId;
+  final int _text2Id = SelectableRegionState.nextSelectableId;
+  final int _text3Id = SelectableRegionState.nextSelectableId;
+
+  Map<int, TextSpan> dataSourceMap = <int, TextSpan>{};
+  Map<int, TextSpan> bulletSourceMap = <int, TextSpan>{};
+  late final Map<int, TextSpan> originSourceData;
+  late final Map<int, TextSpan> originBulletSourceData;
+
+  @override
+  void initState() {
+    super.initState();
+    _initData();
+  }
+
+  @override
+  void dispose() {
+    _selectionController.dispose();
+    super.dispose();
+  }
+
+  void _initData() {
+    for (int i = 1; i <= 7; i += 1) {
+      final int currentSelectableId = SelectableRegionState.nextSelectableId;
+      bulletSourceMap[currentSelectableId] = TextSpan(text: '• Bullet $i');
+    }
+    dataSourceMap[_text1Id] = TextSpan(
+      text: 'This is some bulleted list:\n',
+      children: <InlineSpan>[
+        WidgetSpan(
+          child: Column(
+            children: <Widget>[
+              for (final MapEntry<int, TextSpan> entry in bulletSourceMap.entries)
+                Padding(
+                  padding: const EdgeInsets.only(left: 20.0),
+                  child: Text.rich(
+                    bulletSourceMap[entry.key]!,
+                    selectableId: entry.key,
+                  ),
+                )
+            ],
+          ),
+        ),
+      ],
+    );
+    dataSourceMap[_text2Id] = const TextSpan(
+      text: 'This is some text in a text widget.',
+      children: <InlineSpan>[TextSpan(text: ' This is some more text in the same text widget.')],
+    );
+    dataSourceMap[_text3Id] = const TextSpan(text: 'This is some text in another text widget.');
+    // Save the origin data so we can revert our changes.
+    originSourceData = <int, TextSpan>{ ...dataSourceMap };
+    originBulletSourceData = <int, TextSpan>{ ...bulletSourceMap };
+  }
+
+  void _emphasizeText(List<SelectedContentController<Object>>? controllers, { Map<int, TextSpan>? dataMap }) {
```

Comment:
It looks like this optionally operates on dataMap, if it's provided, and otherwise operates on the instance variable dataSourceMap. Maybe make it always take a dataMap parameter and just pass the one that you want it to operate on?

Reply:
sgtm